python/pyspark/tests.py

```diff
@@ -874,6 +874,15 @@ def test_sortByKey_uses_all_partitions_not_only_first_and_last(self):
             for size in sizes:
                 self.assertGreater(size, 0)
 
+    def test_pipe_functions(self):
+        data = ['1', '2', '3']
+        rdd = self.sc.parallelize(data)
+        with QuietTest(self.sc):
+            self.assertRaises(Py4JJavaError, rdd.pipe('cc').collect)
+        result = rdd.pipe('cat').collect()
```

Comment:
I can do although I think a non-zero code from grep will raise an Exception in the scala implementation. See (that I just found):
https://github.com/apache/spark/blob/master/core/src/test/scala/org/apache/spark/rdd/PipedRDDSuite.scala

I see that grep will return 1 if it doesn't match any lines in a partition. Raising an exception in this case may well be annoying but other shell functions can return 1 on errors (for example http://www.postgresql.org/docs/8.3/static/app-psql.html). I wonder what the best solution is here?